core/mutex.c

```diff
@@ -194,6 +211,9 @@ void mutex_unlock(mutex_t *mutex)
         sched_change_priority(owner, mutex->owner_original_priority);
     }
 #endif
+#if IS_USED(MODULE_CORE_MUTEX_DEBUG)
+    mutex->owner_calling_pc = 0;
```

Comment:
Ah no - if another thread was waiting on the mutex, it will be locked again here, but we don't have the PC of that one.

Not sure if
```suggestion
    mutex->owner_calling_pc = cpu_get_caller_pc();
```

would be any better of if we should just document that. (Or can we get the PC of the blocked thread?)

Reply:
Should be fixed now. There is still a brief time window between clearing this and when the new owner updates this in which theoretically a higher priority thread could wake up and try to take the mutex and find a value of `0` here, but that should barely ever happen. I think for a debug print this is good enough and the issue is now documented.